connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/ClientE2ETestSuite.scala

```diff
@@ -863,6 +866,46 @@ class ClientE2ETestSuite extends RemoteSparkSession with SQLHelper {
     }.getMessage
     assert(message.contains("PARSE_SYNTAX_ERROR"))
   }
+
+  test("protobuf functions") {
+    assert(IntegrationTestUtils.isSparkProtobufJarAvailable)
+    // scalastyle:off line.size.limit
+    // If `common.desc` needs to be updated, execute the following command to regenerate it:
+    //  1. cd connector/connect/common/src/main/protobuf/spark/connect
+    //  2. protoc --include_imports --descriptor_set_out=../../../../test/resources/protobuf-tests/common.desc common.proto
+    // scalastyle:on line.size.limit
+    val descPath = {
```

Comment:
I haven't found a suitable way to use `PlanGenerationTestSuite` and `ProtoToParsedPlanTestSuite` to test 4-args `from_protobuf` and `to_protobuf` due one of the parameters is the absolute file path, so add an E2E test case here.

Reply:
Where is that absolute file path coming from?

Reply:
At first, I wrote the following case in `PlanGenerationTestSuite ` to test `from_protobuf` with `descFilePath`:

```
  test("from_protobuf messageClassName options descFilePath") {
    binary.select(
      pbFn.from_protobuf(
        data = fn.col("bytes"),
        messageName = "StorageLevel",
        descFilePath = "file://fake.desc"))
  }
```

but when I run `ProtoToParsedPlanTestSuite `, there is following errors:

```
[info] - from_protobuf_messageClassName_options_descFilePath *** FAILED *** (11 milliseconds)
[info]   org.apache.spark.sql.AnalysisException: [PROTOBUF_DESCRIPTOR_FILE_NOT_FOUND] Error reading Protobuf descriptor file at path: file://fake.desc.
[info]   at org.apache.spark.sql.errors.QueryCompilationErrors$.cannotFindDescriptorFileError(QueryCompilationErrors.scala:3375)
[info]   at org.apache.spark.sql.protobuf.utils.ProtobufUtils$.parseFileDescriptorSet(ProtobufUtils.scala:234)
[info]   at org.apache.spark.sql.protobuf.utils.ProtobufUtils$.buildDescriptor(ProtobufUtils.scala:212)
[info]   at org.apache.spark.sql.protobuf.utils.ProtobufUtils$.buildDescriptor(ProtobufUtils.scala:149)
[info]   at org.apache.spark.sql.protobuf.ProtobufDataToCatalyst.messageDescriptor$lzycompute(ProtobufDataToCatalyst.scala:57)
[info]   at org.apache.spark.sql.protobuf.ProtobufDataToCatalyst.messageDescriptor(ProtobufDataToCatalyst.scala:56)
[info]   at org.apache.spark.sql.protobuf.ProtobufDataToCatalyst.dataType$lzycompute(ProtobufDataToCatalyst.scala:42)
[info]   at org.apache.spark.sql.protobuf.ProtobufDataToCatalyst.dataType(ProtobufDataToCatalyst.scala:41)
...
[info]   Cause: java.io.FileNotFoundException: file:/fake.desc (No such file or directory)
...
```

`ProtobufDataToCatalyst.dataType` need the `descFilePath` file to resolve `dataType`, so I change the case as follows:

```
  private lazy val descPath = {
    getWorkspaceFilePath(
      "connector",
      "connect",
      "common",
      "src",
      "test",
      "resources",
      "protobuf-tests",
      "common.desc")

  test("from_protobuf messageClassName options descFilePath") {
    binary.select(
      pbFn.from_protobuf(
        data = fn.col("bytes"),
        messageName = "StorageLevel",
        descFilePath = descPath.toFile.getPath))
  }
```

the test will success, but the contens of `.explain` as follows:

```Project [from_protobuf(bytes#0, StorageLevel, Some(/Users/yangjie01/SourceCode/git/spark-mine-sbt/connector/connect/common/src/test/resources/protobuf-tests/common.desc)) AS from_protobuf(bytes)#0]
+- LocalRelation <empty>, [id#0L, bytes#0]
```

the `descFilePath` is a absolute path, any better suggestions?  @hvanhovell

Reply:
find the way to test with `PlanGenerationTestSuite`